### PR TITLE
Improve error handling in proxy 

### PR DESCRIPTION
**Pull request by Arkham**
### WHY are these changes introduced?

Whenever one of the processes connected to the proxy crashes, the CLI crashes as well.

### WHAT is this pull request doing?

Add error handling to the proxy we use for web requests and websockets.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/app/src/cli/utilities/app/http-reverse-proxy.ts [hunk 1]
```diff
@@ -66,18 +66,20 @@ export async function runConcurrentHTTPProcessesAndPathForwardTraffic({
...
 ${outputToken.json(JSON.stringify(rules))}
 `)
 
   const proxy = httpProxy.createProxy()
   const server = http.createServer(function (req, res) {
     const target = match(rules, req)
-    if (target) return proxy.web(req, res, {target})
+    if (target) {
+      return proxy.web(req, res, {target}, (err) => {
+        outputWarn(`Error forwarding web request: ${err}`)
```
Comment (isaacroldan) [inline packages/app/src/cli/utilities/app/http-reverse-proxy.ts:80]: For ruby apps, does this mean we are almost always going to show this warning while the app is starting up?

Comment (Arkham) [reply packages/app/src/cli/utilities/app/http-reverse-proxy.ts:80]: Yup!

**Review by isaacroldan — APPROVED** [review]

File: packages/app/src/cli/utilities/app/http-reverse-proxy.ts (hunk 1 above)
Comment (pepicrft) [reply packages/app/src/cli/utilities/app/http-reverse-proxy.ts:80]: What about having state scoped to the target that indicates whether it's running or not. Then we can use it to:
- Output it as a `debug` (when we pass `--verbose`) if the server is starting.
- Output it  as a warning always if the server was started and all of a sudden we started getting errors.

Comment (Arkham) [reply packages/app/src/cli/utilities/app/http-reverse-proxy.ts:80]: makes sense, i'd also like to get it rendered with the other processes inside `renderConcurrent`

**Pull request merged**